src/main.ts

```diff
@@ -88,19 +89,19 @@ export default class FileHider extends Plugin {
 	/*
 	Enables/Disables the file visibility based. (gets the stylesheet if needed)
 	*/
-	toggleVisibility() {
+	async toggleVisibility() {
 		this.settings.hidden = !this.settings.hidden;
 		for (const path of this.settings.hiddenList) {
 			changePathVisibility(path, this.settings.hidden);
-		};
-		this.saveSettings();
+		}
+		await this.saveSettings();
 	};
 
-	unhidePath(path: string) {
+	async unhidePath(path: string) {
 		let i = this.settings.hiddenList.indexOf(path);
 		this.settings.hiddenList.splice(i, 1);
 		changePathVisibility(path, false);
-		this.saveSettings();
+		await this.saveSettings();
```

Comment:
This and the same call on line 97 don't need to be awaited, they are able to be completed in the background.

Reply:
Well, I have a warning for this because saveSettings should be awaited